C#: Add Hot Reload section for C# Dev Kit (#6828)
microsoft/vscode-docs: docs/csharp/cs-dev-kit-faq.md

### I am not getting whole line completions

Whole line completions are disabled when the [GitHub Copilot](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) extension is enabled to allow you to take advantage of the more advanced [AI completion](/docs/editor/github-copilot.md) capabilities. You can verify that Copilot is enabled by checking if the Copilot logo is present in the lower right corner of VS Code.

> ## Hot Reload

Reviewer: >## [](http://example.com/codeflow?start=0&length=2)

Should we add something like "what types of edits are supported?" and link to [https://learn.microsoft.com/en-us/visualstudio/debugger/supported-code-changes-csharp?view=vs-2022](https://learn.microsoft.com/en-us/visualstudio/debugger/supported-code-changes-csharp?view=vs-2022)

### Hot Reload icon not appearing after I started debugging

> The debugger only starts a Hot Reload session if the option for Hot Reload is enabled in the C# Dev Kit [debugger settings](/docs/csharp/debugging#user-settings). If the option is enabled, it is expected that the Hot Reload icon appear in the bottom bar when debugging:

Author: ```suggestion
The debugger only starts a Hot Reload session if the option for Hot Reload is enabled in the C# Dev Kit [debugger settings](/docs/csharp/debugging#user-settings). If the option is enabled, it is expected that the Hot Reload icon appear in the status bar when debugging:
```

![Hot Reload icon in the bottom bar](images/debugging/hotreload-statusbar.png)

> You can either click on it or can see diagnostic information by opening the **C# Hot Reload** Output window. If you don't see none of those, there is a chance that the project is not supported by C# Dev Kit extension, see [Hot Reload supported projects](/docs/csharp/debugging#user-settings).

Reviewer 2: ```suggestion
You can either click on the Hot Reload icon or can see diagnostic information by opening the **C# Hot Reload** Output window. If you don't see either of those, the project may not be supported by C# Dev Kit extension, see [Hot Reload supported projects](/docs/csharp/debugging#supported-.net-apps-and-scenarios).
```

### I am not getting whole line completions

Whole line completions are disabled when the [GitHub Copilot](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) extension is enabled to allow you to take advantage of the more advanced [AI completion](/docs/editor/github-copilot.md) capabilities. You can verify that Copilot is enabled by checking if the Copilot logo is present in the lower right corner of VS Code.

## Hot Reload

### Hot Reload icon not appearing after I started debugging

> 

Reviewer 3: add .md so /docs/csharp/debugging.md#user-settings

The debugger only starts a Hot Reload session if the option for Hot Reload is enabled in the C# Dev Kit [debugger settings](/docs/csharp/debugging#user-settings). If the option is enabled, it is expected that the Hot Reload icon appears in the status bar while debugging:

![Hot Reload icon in the bottom bar](images/debugging/hotreload-statusbar.png)

> You can either click on the Hot Reload icon or can see diagnostic information by opening the **C# Hot Reload** Output window. If you don't see either of those, the project may not be supported by C# Dev Kit extension, see [Hot Reload supported projects](/docs/csharp/debugging#supported-.net-apps-and-scenarios).

Reviewer 3: The period in .NET will be removed in the header link so 

/docs/csharp/debugging.md#supported-net-apps-and-scenarios

also adding the .md file extension in the relative path above

Author: I wasn't aware of that, thanks for all the pointers!

### What types of edits are supported with Hot Reload?

> See [Supported code changes](https://learn.microsoft.com/en-us/visualstudio/debugger/supported-code-changes-csharp?view=vs-2022) for a complete list of the supported C# code changes with Hot Reload.

Reviewer 3: remove /en-us/
